### PR TITLE
chore(repo): ensure certain sdk docs cannot be merged without running build

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
Should prevent this type of regression https://github.com/winglang/wing/pull/3809

It is still difficult for contributors to realize that these docs are sourced from wingsdk. There are a few possible options to make this easier, but for now I want to at least prevent `main` from being broken.

Misc changes: 
- Prevent the title validation workflow from running on merge queue PRs. They always fail and are not used to determine mergeability.
- Changed the message on PRs when entered into the merge queue, as the previous one was not really accurate

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by WeepingClown13** [comment]
Nice conditional check on the docs! I'm not understanding the mergify comment in line 20 how much I read though 😅

File: .mergify.yml [hunk 1]
```diff
@@ -5,14 +5,19 @@ queue_rules:
...
+              - -files~=^docs/docs/04-standard-library/[^/]+/[^/]+\.md$
 
 pull_request_rules:
   - name: automatic merge
     actions:
       comment:
-        message: Thank you for contributing! Your pull request will be updated from {{ base }} and then merged automatically (do not update manually, and be sure to [allow changes to be pushed to your fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)).
+        message: |
+          Thank you for contributing! If this branch is up-to-date, it will be merged automatically.
+          If not, a new PR will be created with this latest merged into your changes to verify. This PR will be merged after that check is successful.
```
Comment (MarkMcCulloh) [inline .mergify.yml:20]:
```suggestion
          Thank you for contributing! This PR will now be added to the [merge queue](https://mergify.com/merge-queue), or immediately merged to main if up-to-date and the queue is empty.
```

**Comment by MarkMcCulloh** [comment]
@WeepingClown13 That's good feedback! It definitely was clumsy, I was trying to keep it short but I agree it was confusing. 
I changed the wording and simply linked to an explanation of the merge queue instead, let me know what you think about it.

**Comment by WeepingClown13** [comment]
> @WeepingClown13 That's good feedback! It definitely was clumsy, I was trying to keep it short but I agree it was confusing. 
> I changed the wording and simply linked to an explanation of the merge queue instead, let me know what you think about it.

Simple and to the point now, I like it! Though do you think maybe changing part of it to something like "if the base branch is up-to-date" instead of "if up-to-date" is worth it? PR being up-to-date does make sense so I'm not adamant about this change but I also feel something missing. Nonetheless, it is far better now and very much understandable for sure.

File: .mergify.yml [hunk 2]
```diff
@@ -5,14 +5,18 @@ queue_rules:
...
+              # These files must be generated via wingsdk build
+              - -files~=^docs/docs/04-standard-library/[^/]+/[^/]+\.md$
 
 pull_request_rules:
   - name: automatic merge
     actions:
       comment:
-        message: Thank you for contributing! Your pull request will be updated from {{ base }} and then merged automatically (do not update manually, and be sure to [allow changes to be pushed to your fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)).
+        message: |
+          Thank you for contributing! This PR will now be added to the [merge queue](https://mergify.com/merge-queue), or immediately merged to main if up-to-date and the queue is empty.
```
Comment (MarkMcCulloh) [inline .mergify.yml:19]:
```suggestion
          Thanks for contributing, @{{author}}! This PR will now be added to the [merge queue](https://mergify.com/merge-queue), or immediately merged if @{{head}} is up-to-date with @{{base}} and the queue is empty.
```

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**